### PR TITLE
Add Crosswalk API Gateway multi-language examples

### PR DESCRIPTION
**Pull request by danielrbradley**
Create updated examples for AWS API Gateway to go along with updating the main guide (https://github.com/pulumi/pulumi-hugo/pull/867).

- Create comparible examlpes in Typescript, Python and Go
- Create static (S3), Lambda and HTTP proxy routes
- Create Cognito and Lambda authorizers
- All examples and instructions tested locally

### CODE REVIEW

File: aws-apigateway-go-routes/handler/handler.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package main
```
Comment (mikhailshilkov) [inline aws-apigateway-go-routes/handler/handler.go:1]: We probably want to add copyright to all code files as in https://github.com/pulumi/examples/blob/master/azure-go-aks/main.go#L1

File: aws-apigateway-py-routes/__main__.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import pulumi
+import pulumi_aws as aws
+import pulumi_aws_apigateway as apigateway
+import lambdas_
```
Comment (mikhailshilkov) [inline aws-apigateway-py-routes/__main__.py:4]: Is this trailing underscore required in the file name?

File: aws-apigateway-go-routes/main.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			Runtime: lambda.RuntimeGo1dx,
+			Code: pulumi.NewAssetArchive(map[string]interface{}{
+				".": pulumi.NewFileArchive("./bin/handler.zip"),
+			}),
+			Handler: pulumi.String("handler"),
+			Role:    role.Arn,
+		}, pulumi.DependsOn([]pulumi.Resource{logPolicy}))
+		if err != nil {
+			return err
+		}
```
Comment (mikhailshilkov) [inline aws-apigateway-go-routes/main.go:71]: Would it make sense to move lambdas to a separate file/function similar to Python?

**Review by mikhailshilkov — COMMENTED** [review]
These look great to me. Is C# coming?

File: aws-apigateway-py-routes/__main__.py (hunk 2 above)
Comment (danielrbradley) [reply aws-apigateway-py-routes/__main__.py:4]: Seemed like it broke without it - think lambda is a reserved keyword in python.

Comment (mikhailshilkov) [reply aws-apigateway-py-routes/__main__.py:5]: But not `lambdas` I guess?

Comment (danielrbradley) [reply aws-apigateway-py-routes/__main__.py]: Yup - the suffix can be removed fine 👍

File: aws-apigateway-go-routes/main.go [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+		apiId := restAPI.Api.ApplyT(func(api *awsapigateway.RestApi) pulumi.StringOutput {
+			return api.ID().ToStringOutput()
+		}).ApplyT(func(id interface{}) string {
+			return id.(string)
+		}).(pulumi.StringOutput)
+		stageName := restAPI.Stage.ApplyT(func(stage *awsapigateway.Stage) pulumi.StringOutput {
+			return stage.StageName
+		}).ApplyT(func(stageName interface{}) string {
+			return stageName.(string)
+		}).(pulumi.StringOutput)
```
Comment (mikhailshilkov) [inline aws-apigateway-go-routes/main.go:148]: This double-apply looks a bit wild. Aren't we getting the stageName output from the first ApplyT? Same question to the `apiId` above.

**Review by mikhailshilkov — APPROVED** [review]
LGTM with one question

File: aws-apigateway-go-routes/main.go (hunk 4 above)
Comment (danielrbradley) [reply aws-apigateway-go-routes/main.go:148]: Yes it does appear to be required. The first apply retrieve the sub-property, which itself is an output needing to be resolved. The second apply is to allow us to convert the unknown value to the correct type.

**Comment by lukehoban** [comment]
🎉

**Pull request merged**